### PR TITLE
fix: schematics cache

### PR DESCRIPTION
**Pull request by cpaulve-1A**
Remove incorrect outputs and dependency on project.json

Only package with non standard structure should have custom outputs and dependsOn properties

Remove the reference in o3r/styling/tsconfig.builders.json to the src folder (composite build)

o3r/styling builders have a dependency over the compiled package and should not try to compile it themselves

### CODE REVIEW

**Review by jbourgeois-1A — DISMISSED** [review]

File: nx.json [hunk 1]
```diff
@@ -181,6 +178,9 @@
         "{projectRoot}/dist/middlewares/**/*.js",
         "{projectRoot}/dist/middlewares/**/*.d.ts",
         "{projectRoot}/dist/middlewares/**/*.d.ts.map",
+        "{projectRoot}/dist/ng-add/**/*.js",
```
Comment (kpanot) [inline nx.json:181]: this `ng-add` should not be in the `dist/` folder directly but within the `schematics/` folder.
Do we have exception in the code?

Comment (kpanot) [inline nx.json:181]: this `ng-add` should not be in the `dist/` folder directly but within the `schematics/` folder.
Do we have exception in the code?

File: nx.json [hunk 2]
```diff
@@ -134,10 +134,7 @@
         "^source"
       ],
       "outputs": [
-        "{projectRoot}/dist/",
-        "!{projectRoot}/dist/schematics",
-        "!{projectRoot}/dist/builders",
-        "!{projectRoot}/dist/middlewares"
+        "{projectRoot}/dist/"
```
[the remark this answers is not in the document]
Comment (cpaulve-1A) [reply nx.json:137]: actually, we are not sure the ! is supported, I can leave it but when we started on nx this was not supported

File: nx.json (hunk 1 above)
Comment (cpaulve-1A) [reply nx.json:181]: I think we had when I started this, but I can't find it anywhere

File: nx.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (kpanot) [reply nx.json:137]: After your test it seems to not be supported.
As the regexp are now supported (cf https://github.com/nrwl/nx/pull/10894), we can try a `["{projectRoot}/dist/!(schematics|builders|middlewares)/**", "{projectRoot}/dist/!(schematics|builders|middlewares)" ]` ?

(or maybe only `["{projectRoot}/dist/!(schematics|builders|middlewares)" ]`)

[the remark this answers is not in the document]
Comment (cpaulve-1A) [reply nx.json:137]: it seems to have an impact on our build, let's keep it for now

**Review by kpanot — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by vscaiceanu-1a — APPROVED** [review]

**Pull request merged**